integration_tests/mockApis/premises.ts

```diff
@@ -90,17 +90,17 @@ const stubPremisesStaffMembers = (args: { premisesId: string; staffMembers: Arra
 const stubPremiseCapacity = (args: {
   premisesId: string
   startDate: string
-  endDate: string
+  endDate?: string
   excludeSpaceBookingId?: string
-  premiseCapacity: Cas1PremiseCapacity
+  premiseCapacity?: Cas1PremiseCapacity
 }) =>
   stubFor({
     request: {
       method: 'GET',
       urlPath: paths.premises.capacity({ premisesId: args.premisesId }),
       queryParameters: {
-        startDate: { equalTo: args.startDate },
-        endDate: { equalTo: args.endDate },
+        startDate: args.startDate ? { equalTo: args.startDate } : undefined,
+        endDate: args.endDate ? { equalTo: args.endDate } : undefined,
```

Comment:
Are these changes needed? Seems the changes in the signature clash with the changes in the expected params (only `endDate` is now marked as optional in signature, yet both `endDate` and `startDate` check for truthiness), and I think we always do pass the `premisesCapacity` too?

Reply:
So, this change is about allowing the stubbing to be less specific.
All the parameters are always passed in the API call, but it was getting quite complex to derive the exact parameters (startDate, endDate) that need to be matched by the mock.
I guess the thing is that if we don't specify the mock's query parameters, we should check that the mock was called with the parameters we expect - which is much the same amount of effort.

Anyway, I've reverted the mock above and changed the code to supply the correct start and end dates in all cases.

Reply:
I think it does make sense, and it's most likely useful to have a more flexible mock -- I was mostly confused that only `endDate` in the signature was made optional, and that the response would be `undefined` if capacity wasn't provided -- I'd expect that response to be a factory if not defined, at least?